### PR TITLE
Add Int32 support for gdal driver

### PR DESCRIPTION
**Pull request by cpaulik**
Our Int32 geotiffs have been failing to render correctly. 

This fixes the problem for us.

Should I open a separate MR for the 3.0.x branch?
Do you require a visual test for this kind of change?

### CODE REVIEW

File: plugins/input/gdal/gdal_featureset.cpp [hunk 1]
```diff
@@ -397,6 +397,25 @@ feature_ptr gdal_featureset::get_feature(mapnik::query const& q)
...
                 break;
             }
+            case GDT_Int32:
+            {
+                mapnik::image_gray32s image(im_width, im_height);
+                image.set(std::numeric_limits<std::int32_t>::max());
+                raster_nodata = band->GetNoDataValue(&raster_has_nodata);
+                raster_io_error = band->RasterIO(GF_Read, x_off, y_off, width, height,
+                                                 image.data(), image.width(), image.height(),
+                                                 GDT_Int32, 0, 0, &psExtraArg);
```
Comment (artemp) [inline plugins/input/gdal/gdal_featureset.cpp]: `psExtraArg` is not declared anywhere /cc @cpaulik

Comment (cpaulik) [reply plugins/input/gdal/gdal_featureset.cpp]: Oops. I left that in from https://github.com/mapnik/mapnik/pull/3975. Will take it out

Comment (artemp) [reply plugins/input/gdal/gdal_featureset.cpp]: :+1:

**Comment by cpaulik** [comment]
I've opened 

- https://github.com/mapnik/test-data-visual/pull/83
- https://github.com/mapnik/test-data/pull/21

which include the testdata and the rendering test for a int32 file.

**Comment by cpaulik** [comment]
Updated merge request with new submodule commits. Let's hope CI passes.

**Comment by artemp** [comment]
:crossed_fingers: :+1:

**Pull request merged**